packages/rum-core/src/error-logging/error-logging.js

```diff
@@ -110,12 +110,31 @@ class ErrorLogging {
     }
   }
 
-  registerGlobalEventListener() {
+  onUnhandledError() {
     window.addEventListener('error', errorEvent =>
       this.logErrorEvent(errorEvent)
     )
   }
 
+  onUnhandledRejection() {
+    window.addEventListener('unhandledrejection', promiseRejectionEvent =>
+      this.logPromiseEvent(promiseRejectionEvent)
+    )
+  }
+
+  logPromiseEvent(promiseRejectionEvent) {
+    const prefix = 'Unhandled promise rejection: '
+    const { reason } = promiseRejectionEvent
+
+    if (reason == null) {
+      this.logError(prefix + '<no reason defined>')
+    } else if (typeof reason === 'object') {
+      this.logError(reason)
```

Comment:
`reason` could be any type of object, but logError only accepts string or Error .

Reply:
The only interesting things that we need from the object is `message` and `stack`. logError handles that case. I will make it clear here.